File: java/Produto.java
public class Produto {

    private int id;

    private String nome;

    private double preco;

    private int estoque;

    public Produto(
            int id,
            String nome,
            double preco,
            int estoque
    ) {

        this.id = id;

        this.nome = nome;

        this.preco = preco;

        this.estoque = estoque;
    }

    public int getId() {

        return id;
    }

    public String getNome() {

        return nome;
    }

    public double getPreco() {

        return preco;
    }

    public int getEstoque() {

        return estoque;
    }

    public void reduzirEstoque() {

        if (estoque > 0) {

            estoque--;
        }
    }

    public void exibirProduto() {

        System.out.println(
                "ID: " + id
        );

        System.out.println(
                "Produto: " + nome
        );

        System.out.println(
                "Preço: R$ " +
                String.format("%.2f", preco)
        );

        System.out.println(
                "Estoque: " + estoque
        );
    }
}

File: backend/models/Produto.php
<?php

class Produto
{
    private int $id;

    private string $nome;

    private float $preco;

    private int $estoque;

    public function __construct(
        int $id,
        string $nome,
        float $preco,
        int $estoque
    ) {
        $this->id = $id;
        $this->nome = $nome;
        $this->preco = $preco;
        $this->estoque = $estoque;
    }

    public function getId(): int
    {
        return $this->id;
    }

    public function getNome(): string
    {
        return $this->nome;
    }

    public function getPreco(): float
    {
        return $this->preco;
    }

    public function getEstoque(): int
    {
        return $this->estoque;
    }

    public function reduzirEstoque(): void
    {
        if ($this->estoque > 0) {
            $this->estoque--;
        }
    }

    public function exibirProduto(): array
    {
        return [
            'id' => $this->id,
            'nome' => $this->nome,
            'preco' => $this->preco,
            'estoque' => $this->estoque
        ];
    }
}